src/ipcache.cc

```diff
@@ -702,7 +702,7 @@ ipcache_init(void)
     ipcache_low = (long) (((float) Config.ipcache.size *
                            (float) Config.ipcache.low) / (float) 100);
     n = hashPrime(ipcache_high / 4);
-    ip_table = hash_create((HASHCMP *) strcmp, n, hash4);
+    ip_table = hash_create((HASHCMP *) strcasecmp, n, hash4);
```

Comment:
The case-sensitive hash4() function cannot be used in a case-insensitive hash. Please replace the hash4 parameter with a name of a new HASHHASH function that yields the same hash value for all c-strings that differ only in case. Please let me know if you need help with an efficient implementation of such a hash function.

Background: One of the invariants most hash implementations must observe is that equal items must yield an equal hash. Since we do not want to cache 1000 case variations of the same domain name and risk triggering various assertions, we have to keep the two functions in sync.

FWIW, the same principle applies to [std::unordered_map](https://en.cppreference.com/w/cpp/container/unordered_map): "If two keys are equivalent, the hash function must return the same value for both keys."

Coincidentally, we just fixed a similar (but "opposite") de-synchronization bug in another Squid hash -- see commit b91a855.